app/docs/src/reference/code-generation.md

```diff
@@ -0,0 +1,111 @@
+# Code Generation Functions
+
+Code Generation functions are a kind of [function](./function.md) which generate
+code or configuration from [component](./components.md) data. They transform
+your component's properties into the exact format needed by external tools and
+APIs, making it easy to see what will be sent to your cloud provider before
+actions execute.
+
+## How Code Generation Functions Work
+
+Code Generation functions take a component's current configuration and generate
+formatted output - typically JSON, YAML, or other structured formats. The
+generated code:
+
+- Appears in the Code Gen section of the component details page
+- Can be accessed by action functions when creating or updating resources
+- Is available in other functions as part of the component's code property
+- Updates automatically when the component's properties change
+
+This provides transparency into exactly what configuration will be sent to
+external services and allows you to validate it before taking action.
+
+## When Code Generation Functions Run
+
+Code Generation functions execute automatically:
+
+- When a component is first created
+- When component properties that the function depends on change
+- Before actions that use the generated code execute
+
+The generated code is stored with the component, so action functions can access
+it without re-running the code generation.
+
+## Code Generation Function Arguments
+
+Code Generation functions receive a single `component` argument with these
+properties:
+
+- `domain`: The component's domain properties (your configuration)
+- `resource`: The resource information from previous actions (if any)
+- `deleted_at`: A timestamp if the component has been deleted
+
+This gives the function access to both the intended configuration and any
+current resource state.
+
+## Code Generation Function Return Value
+
+Code Generation functions return an object with two properties:
+
+```typescript
+{
+    format: "json",  // or "yaml", "toml", etc.
+    code: '{ "poop": "canoe" }',  // the generated code as a string
+}
+```
```

Comment:
I like this with the comments. Good call on using `typescript` since `json` does not support comments of course